src/ProfileEndpoints.jl

```diff
@@ -117,17 +130,46 @@ function _start_cpu_profile(n, delay)
     return resp
 end
 
-function _cpu_profile_response(filename; with_pprof::Bool)
+function handle_cpu_profile_stop(qp, stage_path = nothing)
+    with_pprof = parse(Bool, get(qp, "pprof", default_pprof()))
+    path = "cpu_profile"
+    if stage_path === nothing
+        # Defer the potentially expensive profile symbolication to a non-interactive thread
+        return fetch(Threads.@spawn _cpu_profile_get_response($path; with_pprof=$with_pprof))
+    end
+    path = joinpath(stage_path, "cpu_profile")
+    # Defer the potentially expensive profile symbolication to a non-interactive thread
+    return fetch(Threads.@spawn _cpu_profile_get_response_and_write_to_file($path; with_pprof=$with_pprof))
+end
+
+function _cpu_profile_get_response_and_write_to_file(filename; with_pprof::Bool)
+    if with_pprof
+        PProf.pprof(out=filename, web=false)
+        filename = "$filename.pb.gz"
+        return _http_create_response_with_profile_as_file(filename)
+    else
+        iobuf = IOBuffer()
+        data = Profile.retrieve()
+        serialize(iobuf, data)
+        filename = "$filename.profile"
+        open(filename, "w") do io
+            write(io, iobuf.data)
+        end
+        return _http_create_response_with_profile_as_file(filename)
+    end
+end
+
+function _cpu_profile_get_response(filename; with_pprof::Bool)
     if with_pprof
         prof_name = tempname()
         PProf.pprof(out=prof_name, web=false)
         prof_name = "$prof_name.pb.gz"
-        return _http_response(read(prof_name), "$filename.pb.gz")
+        return _http_create_response_with_profile_inlined(read(prof_name), "$filename.pb.gz")
     else
         iobuf = IOBuffer()
         data = Profile.retrieve()
         serialize(iobuf, data)
-        return _http_response(take!(iobuf), "$filename.prof.bin")
+        return _http_create_response_with_profile_inlined(iobuf.data, "$filename.profile")
```

Comment:
I noticed you changed the file extension. Can you expand on why?
I'm not sure that `.prof.bin` was a good choice either...

Maybe it should be `.profile.juliaserialized`? or `.profile.jlserialized` or something?

Reply:
`.profile` seems reasonable to me.

`.profile.juliaserialized` seems overly verbose.